pkg/block/index.go

```diff
@@ -524,6 +548,66 @@ func IgnoreDuplicateOutsideChunk(_, _ int64, last, curr *chunks.Meta) (bool, err
 	return true, nil
 }
 
+// FixDuplicateSamplesOutsideChunk takes in two pointers to chunks.Meta structs,
+// last and curr, and returns a new chunks.Meta struct and an error. If last is
+// nil, the function returns curr and nil. Otherwise, it creates a new empty
+// chunk based on the encoding of curr.Chunk.
+//
+// Parameters:
+// - last: a pointer to the last chunk.Meta struct
+// - curr: a pointer to the current chunk.Meta struct
+//
+// Return:
+// - *chunks.Meta: a pointer to the new chunk.Meta struct
+// - error: an error if any occurred during the function execution.
+func FixDuplicateSamplesOutsideChunk(last, curr *chunks.Meta) (*chunks.Meta, error) {
+	if last == nil {
+		return curr, nil
+	}
+
+	c, err := chunkenc.NewEmptyChunk(curr.Chunk.Encoding())
+	if err != nil {
+		return curr, err
+	}
+	ncurr := &chunks.Meta{
+		MinTime: int64(0),
+		MaxTime: curr.MaxTime,
+		Chunk:   c,
+	}
+	if curr.MinTime == last.MaxTime {
+		var it chunkenc.Iterator
+		it = curr.Chunk.Iterator(it)
+		app, _ := ncurr.Chunk.Appender()
+		typ := it.Next()
+		for ; typ != chunkenc.ValNone; typ = it.Next() {
+			var t int64
+			var v float64
+			switch typ {
+			case chunkenc.ValFloat:
+				t, v = it.At()
+				if t <= last.MaxTime {
+					continue
+				}
+				app.Append(t, v)
+			default:
+				continue
+			}
+			if ncurr.MinTime == 0 {
+				ncurr.MinTime = t
+			}
+		}
+		if ncurr.Chunk.Bytes() == nil {
+			return curr, errors.Errorf("empty chunk")
+		}
+	} else {
+		return curr, nil
+	}
+	if ncurr.MinTime == 0 {
```

Comment:
This seems unnecessary. It can only happen if a sample t is 0, but chunk minT == last chunk maxT. Seems impossible.